src/spec/cli.cr

```diff
@@ -0,0 +1,136 @@
+require "./formatter"
+require "option_parser"
+
+module Spec
+  # :nodoc:
+  class CLI
+    # :nodoc:
+    class Options
+      property formatters = Array(Spec::Formatter).new
+      property locations = Array({String, Int32}).new
+
+      property! default_formatter : Spec::Formatter
+      property! fail_fast : Bool
+      property! line : Int32
+      property! no_color : Bool
+      property! pattern : String
+      property! slowest : Int32
+    end
+
+    getter options : Options
+
+    private getter argv : Array(String)
+    private getter parser = OptionParser.new
+    private getter stderr : IO
+    private getter stdout : IO
+
+    private getter? prepared : Bool = false
+
+    def initialize(@argv, *, @options = Options.new, @stderr = STDERR, @stdout = STDOUT)
+      setup
+    end
+
+    def prepare
+      return if prepared?
+
+      parser.parse(argv)
+
+      unless argv.empty?
+        terminate "Error: unknown argument '#{argv.first}'"
+      end
+
+      if ENV["SPEC_VERBOSE"]? == "1"
+        options.default_formatter = Spec::VerboseFormatter.new
+      end
+    ensure
+      @prepared = true
+    end
+
+    def run
+      prepare unless prepared?
```

Comment:
Don't need the condition here, will be checked in the method anyway.

Reply:
Ah, was a left over from previous tests. Will remove. Thank you.